src/ppom-controller.test.ts

```diff
@@ -0,0 +1,202 @@
+      const spy = buildFetchSpy({
+        status: 200,
+        json: () => [
+          ...VERSION_INFO,
+          {
+            name: 'data',
+            chainId: '0x2',
+            version: '1.0.3',
+            checksum:
+              '409a7f83ac6b31dc8c77e3ec18038f209bd2f545e0f4177c2e2381aa4e067b49',
+            filePath: 'data',
+          },
+        ],
+      });
+
+      await ppomController.use(async () => {
+        return Promise.resolve();
+      });
+
+      expect(spy).toHaveBeenCalledTimes(3);
+    });
+
+    it('should throw error', async () => {
+      buildFetchSpy({
+        status: 500,
+      });
+
+      await expect(async () => {
+        await ppomController.use(async () => {
+          return Promise.resolve();
+        });
+      }).rejects.toThrow('Failed to fetch version info');
+    });
+
+    it('should throw error if fetch for blob return 500', async () => {
+      buildFetchSpy(undefined, {
+        status: 500,
+      });
+
+      await expect(async () => {
+        await ppomController.use(async () => {
+          return Promise.resolve();
+        });
+      }).rejects.toThrow('Failed to fetch file data');
+    });
+
+    it('should refresh data if refreshInterval is passed', async () => {
+      const spy = buildFetchSpy();
+
+      await ppomController.use(async () => {
+        return Promise.resolve();
+      });
+      expect(spy).toHaveBeenCalledTimes(3);
+      await ppomController.use(async () => {
+        return Promise.resolve();
+      });
+      expect(spy).toHaveBeenCalledTimes(3);
+
+      ppomController.setRefreshInterval(0);
+      await ppomController.use(async () => {
+        return Promise.resolve();
+      });
+      expect(spy).toHaveBeenCalledTimes(4);
+    });
+
+    it('should refresh data if network is changed', async () => {
+      let callBack: any;
+      ppomController = new PPOMController({
+        storageBackend: storageBackendReturningData,
+        provider: { sendAsync: Promise.resolve() },
+        chainId: '0x1',
+        onNetworkChange: (func: any) => {
+          callBack = func;
+        },
+      });
+      const spy = buildFetchSpy({
+        status: 200,
+        json: () => [
+          ...VERSION_INFO,
+          {
+            name: 'data',
+            chainId: '0x2',
+            version: '1.0.3',
+            checksum:
+              '409a7f83ac6b31dc8c77e3ec18038f209bd2f545e0f4177c2e2381aa4e067b49',
+            filePath: 'data',
+          },
+        ],
+      });
+
+      await ppomController.use(async () => {
+        return Promise.resolve();
+      });
+      expect(spy).toHaveBeenCalledTimes(3);
+      await ppomController.use(async () => {
+        return Promise.resolve();
+      });
+      expect(spy).toHaveBeenCalledTimes(3);
+
+      callBack('0x2');
+      await ppomController.use(async () => {
+        return Promise.resolve();
+      });
+      expect(spy).toHaveBeenCalledTimes(4);
+    });
+  });
+
+  describe('PPOM', () => {
```

Comment:
I'm not sure what this grouping means as we traditionally group the tests using the public methods.

Would we want describes and tests for `updatePPOM`, `use`, and `setRefreshInterval`?

Reply:
I changed grouping.